week_2/pet.rb

```diff
@@ -110,6 +115,6 @@ def self.cost_to_feed(pets, days)
   #
   # Note - Order is not important
   def self.group_by_animal_type(pets)
-    raise NotImplementedError # TODO
+    pets.group_by { |pet| pet.animal_type_id }
```

Comment:
This will return an hash whose value is an array or `Pet`. But you have to return hash with key as `animal_type_id` and value as array of pet name.
```suggestion
    pet_group = pets.group_by { |pet| pet.animal_type_id }
    res = {}
    pet_group.each do |id, pet_arr|
      res[id] = pet_arr.map { |p| p.name| }
    end
    
    res
```
https://www.rubyguides.com/2018/10/ruby-map-method/